tasks/configure.yml

```diff
@@ -122,6 +122,54 @@
   register: postgresql_configuration_pt3
   changed_when: "'_OK_' not in postgresql_configuration_pt3.stdout"
 
+- name: PostgreSQL | Generate SSL Private Key | RedHat
+  openssl_privatekey:
+    path: "{{ postgresql_ssl_key_file }}"
+    type: "RSA"
+    size: "{{ postgresql_ssl_key_length }}"
+  become: true
+  become_user: "{{ postgresql_service_user }}"
+  when: ansible_os_family == "RedHat" and
+        postgresql_ssl == "on"
+
+- name: PostgreSQL | Secure SSL Private Key | RedHat
+  file:
+    path: "{{ postgresql_ssl_key_file }}"
+    owner: "{{ postgresql_service_user }}"
+    group: "{{ postgresql_service_group }}"
+    mode: 0600
+  when: ansible_os_family == "RedHat" and
+        postgresql_ssl == "on"
+
+- name: PostgreSQL | Create CSR for SSL Certificate | RedHat
+  openssl_csr:
+    common_name: "{{ ansible_fqdn }}"
+    path: "{{ postgresql_ssl_csr_file }}"
+    privatekey_path: "{{ postgresql_ssl_key_file }}"
+  become: true
+  become_user: "{{ postgresql_service_user }}"
+  when: postgresql_ssl == "on"
```

Comment:
Did you forget to add `ansible_os_family` constraint?

Reply:
add this and it should be good to go.